numba/roc/stubs.py

```diff
@@ -99,6 +99,19 @@ def activelanepermute_wavewidth(src, laneid, identity, useidentity):
     raise _stub_error
 
 
+def ds_permute(src_lane, dest_lane):
```

Comment:
What is "ds"?

Reply:
From the [AMD GCN3 ISA Architecture Manual](http://32ipi028l5q82yhj72224m8j-wpengine.netdna-ssl.com/wp-content/uploads/2016/08/AMD_GCN3_Instruction_Set_Architecture_rev1.1.pdf) I think it's `Data Share`, see section 10.3.3 on `Data Share Indexed`.